storage/libradb/src/lib.rs

```diff
@@ -168,23 +171,40 @@ impl LibraDB {
 
         let path = db_root_path.as_ref().join("libradb");
         let instant = Instant::now();
-        let db = Arc::new(DB::open(path.clone(), cf_opts_map).expect("LibraDB open failed"));
+
+        let db = Arc::new(if readonly {
+            ensure!(path.as_path().is_dir(), "libradb directory is not found.");
+            ensure!(
+                log_dir.is_some(),
+                "Must provide log_dir if opening in readonly mode."
+            );
+            let db_log_dir = log_dir.unwrap().to_str().expect("Invalid directory");
+            info!("log stored at {}", db_log_dir);
```

Comment:
probably roll these into `DB::open_readonly()`?

Reply:
do you think we should have an LibraDB::open_readonly() ?  I didn't do that since the changes are relatively small.

Also, schemadb does't depends on logs, so log checking and directory printing should not really be merged in there.

Reply:
make log_dir a `P` (see other comment) and pass those into `DB::open_readonly` directly, if they fail to convert or are not dirs, Db will complain anyway.